### PR TITLE
Add query for service account performing remote PowerShell

### PR DESCRIPTION
**Pull request by miflower**
All of the details for this query are included inline as comments.

### CODE REVIEW

File: Execution/ServiceAccountsPerformingRemotePS.txt [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	MiscEvents                                                      // ...
+	| where ActionType == 'PowerShellCommand'                       // ...PowerShell commands seen...
+	| where InitiatingProcessFileName =~ 'wsmprovhost.exe'          // ...whose parent was wsmprovhost.exe (RemotePS Server)...
+    | extend AccountName = InitiatingProcessAccountName             // ...and add an AccountName field so the join is easier
+) on AccountName
+// At this point, we have all of the commands that were ran by service accounts
+| extend Command = tostring(parse_json(AdditionalFields).Command)   // Extract the actual PowerShell command that was executed
+| where Command !in (WhitelistedCmdlets)                            // Remove any values that match the whitelisted cmdlets
+| summarize EventTime=max(EventTime), min(EventTime),               // Then group all of the cmdlets and calculate the min/max times of execution...
+    makeset(Command), count(), ReportId=max(ReportId) by            // ...as well as creating a list of cmdlets ran and the count..
```
[the remark this answers is not in the document]
Comment (miflower) [reply Execution/ServiceAccountsPerformingRemotePS.txt]: ReportId is included at it was (not sure if it still is) a required field when creating the scheduled queries.

File: Execution/ServiceAccountsPerformingRemotePS.txt [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+// Now we need to find RemotePS sessions that were spawned by those accounts
+// Note that we look at all powershell cmdlets executed to form a 29-day baseline to evaluate the data on today
+| join kind=rightsemi (                                             // Start by dropping the account name and only tracking the...
+	MiscEvents                                                      // ...
+	| where ActionType == 'PowerShellCommand'                       // ...PowerShell commands seen...
+	| where InitiatingProcessFileName =~ 'wsmprovhost.exe'          // ...whose parent was wsmprovhost.exe (RemotePS Server)...
+    | extend AccountName = InitiatingProcessAccountName             // ...and add an AccountName field so the join is easier
+) on AccountName
+// At this point, we have all of the commands that were ran by service accounts
+| extend Command = tostring(parse_json(AdditionalFields).Command)   // Extract the actual PowerShell command that was executed
```
[the remark this answers is not in the document]
Comment (miflower) [reply Execution/ServiceAccountsPerformingRemotePS.txt]: Added.

File: Execution/ServiceAccountsPerformingRemotePS.txt [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+// At this point, we have a list of accounts that we believe to be service accounts
+// Now we need to find RemotePS sessions that were spawned by those accounts
+// Note that we look at all powershell cmdlets executed to form a 29-day baseline to evaluate the data on today
+| join kind=rightsemi (                                             // Start by dropping the account name and only tracking the...
+	MiscEvents                                                      // ...
+	| where ActionType == 'PowerShellCommand'                       // ...PowerShell commands seen...
+	| where InitiatingProcessFileName =~ 'wsmprovhost.exe'          // ...whose parent was wsmprovhost.exe (RemotePS Server)...
+    | extend AccountName = InitiatingProcessAccountName             // ...and add an AccountName field so the join is easier
+) on AccountName
+// At this point, we have all of the commands that were ran by service accounts
```
[the remark this answers is not in the document]
Comment (miflower) [reply Execution/ServiceAccountsPerformingRemotePS.txt]: I like the simpler join, very nice.  Also added in the ActionType filter
Added.

File: Execution/ServiceAccountsPerformingRemotePS.txt [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+);
+let WhitelistedCmdlets = pack_array(                                // List of whitelisted commands that don't provide a lot of value
+    'prompt',
+    'Out-Default',
+    'out-lineoutput',
+    'format-default',
+    'Set-StrictMode',
+    'TabExpansion2'
+);
+let WhitelistedAccounts = pack_array();                             // List of accounts that are known to perform this activity in the environment and can be ignored
```
[the remark this answers is not in the document]
Comment (miflower) [reply Execution/ServiceAccountsPerformingRemotePS.txt]: Updated the whitelist to contain a single dummy account name in it as the functionality should still be there.

File: Execution/ServiceAccountsPerformingRemotePS.txt [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+// To accomplish this, we utilize LogonEvents and MiscEvents to find cmdlets ran that meet the criteria
+// One of the main advantages of this method is that only requires server telemetry, and not the attacking client
+// The first phase relies on the LogonEvents to determine whether an account is a service account or not, consider the following accounts with logons:
+// random_user has LogonEvents with type 2, 3, 7, 10 & 11
+// random_service_account 'should' only have LogonEvents with type 3,4 or 5
+//
+let InteractiveTypes = pack_array(                                  // Declare Interactive logon type names
+    'Interactive',
+    'CachedInteractive',
+    'Unlock',
```
[the remark this answers is not in the document]
Comment (miflower) [reply Execution/ServiceAccountsPerformingRemotePS.txt]: Added.

File: Lateral Movement/ServiceAccountsPerformingRemotePS.txt [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    'prompt',
+    'Out-Default',
+    'out-lineoutput',
+    'format-default',
+    'Set-StrictMode',
+    'TabExpansion2'
+);
+let WhitelistedAccounts = pack_array('FakeWhitelistedAccount');     // List of accounts that are known to perform this activity in the environment and can be ignored
+LogonEvents                                                         // Get all logon events...
+| where AccountName !in (WhitelistedAccounts)                       // ...where it is not a whitelisted account...
```
Comment (TomerAlpert) [inline Lateral Movement/ServiceAccountsPerformingRemotePS.txt]: Change it to !in~
We lowercase all account names, IIRC, so could compare case sensitive - but still this change would probably save some trouble to some users....

File: Lateral Movement/ServiceAccountsPerformingRemotePS.txt [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+// At this point, we have a list of accounts that we believe to be service accounts
+// Now we need to find RemotePS sessions that were spawned by those accounts
+// Note that we look at all powershell cmdlets executed to form a 29-day baseline to evaluate the data on today
+| join kind=rightsemi (                                             // Start by dropping the account name and only tracking the...
+	MiscEvents                                                      // ...
+	| where ActionType == 'PowerShellCommand'                       // ...PowerShell commands seen...
+	| where InitiatingProcessFileName =~ 'wsmprovhost.exe'          // ...whose parent was wsmprovhost.exe (RemotePS Server)...
+    | extend AccountName = InitiatingProcessAccountName             // ...and add an AccountName field so the join is easier...
+    | extend AccountDomain = InitiatingProcessAccountDomain         // ...and add an AccountDomain field so the join is easier
+) on AccountName, AccountDomain
```
Comment (TomerAlpert) [inline Lateral Movement/ServiceAccountsPerformingRemotePS.txt]: Sorry, let’s remove account domain join for now, still we address the data consistency issue. I’m terribly sorry if my email was not clear on this...

File: Lateral Movement/ServiceAccountsPerformingRemotePS.txt [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+// At this point, we have all of the commands that were ran by service accounts
+| extend Command = tostring(extractjson('$.Command', AdditionalFields))   // Extract the actual PowerShell command that was executed
+| where Command !in (WhitelistedCmdlets)                            // Remove any values that match the whitelisted cmdlets
+| summarize EventTime=max(EventTime), min(EventTime),               // Then group all of the cmdlets and calculate the min/max times of execution...
+    makeset(Command), count(), ReportId=max(ReportId) by            // ...as well as creating a list of cmdlets ran and the count..
```
Comment (TomerAlpert) [inline Lateral Movement/ServiceAccountsPerformingRemotePS.txt]: Why do you calculate max(ReportId)? How is it helpful?

File: Lateral Movement/ServiceAccountsPerformingRemotePS.txt (hunk 6 above)
Comment (miflower) [reply Lateral Movement/ServiceAccountsPerformingRemotePS.txt]: Added, will be in next PR.

File: Lateral Movement/ServiceAccountsPerformingRemotePS.txt (hunk 7 above)
Comment (miflower) [reply Lateral Movement/ServiceAccountsPerformingRemotePS.txt]: Added, will be in next PR.

File: Lateral Movement/ServiceAccountsPerformingRemotePS.txt (hunk 8 above)
Comment (miflower) [reply Lateral Movement/ServiceAccountsPerformingRemotePS.txt]: ReportId is a required field for custom detections.  This query can be dropped straight into a custom detection and run as-is (now that the typo in the whitelisted account is fixed).

**Pull request merged**